docusaurus/docs/how-to-guides/data-source-plugins/add-migration-handler-for-backend.md

```diff
@@ -0,0 +1,177 @@
+
+This guide walks you through the steps necessary to use the latest tooling to automatically migrate queries and avoid duplicating code.
+
+For the last step in this guide, there are two suggested approaches. The first one requires that Grafana to be configured to run data source as standalone API servers, this behavior is currently behind the feature flag [grafanaAPIServerWithExperimentalAPIs](https://github.com/grafana/grafana/blob/3457f219be1c8bce99f713d7a907ee339ef38229/pkg/services/featuremgmt/registry.go#L519) and also requires to run on a Grafana version bigger or equal to 11.4. The second approach requires more boilerplate code but uses stable APIs that can be used in any recent Grafana version.
+
+Also, it’s not the goal of this migration system to support two-way migrations. Only forward migrations are supported. Migrations are not automatically persisted, users need to manually save changes to ensure the migration works as expected. 
+
+# Step 1 (optional): Add a query schema
+
+First of all, plugins don’t need to have strongly typed queries. While this lowers the barrier for plugin development, plugins that don’t define types are harder to scale and maintain. The first step in this guide is to add the required files to define the plugin query.
+
+See the following example: [grafana-plugin-examples#400](https://github.com/grafana/grafana-plugin-examples/pull/400). As you can see, there are multiple files to create. These will be used for both generating OpenAPI documentation and validating that the received queries are valid:
+
+ - `query.go`: This file defines the Golang types for your query. For automatic migrations to work, it’s important that your query extends the new `v0alpha1.CommonQueryProperties`. After that, just define your query custom properties.
+ - `query_test.go`: This test file is both used to check that all the JSON files are up to date with the query model and to generate them. The first time you execute the test, it will generate these files (take into account that query.types.json needs to exist, even if empty).
+ - `query.*.json`: Automatically generated files. These schemas will be used for OpenAPI documentation. This is a feature in progress that is still not available for external plugins.
```

Comment:
```suggestion
 - `query.*.json`: Automatically generated files. These schemas will be used for OpenAPI documentation. This is a feature in progress only available to plugins developed by Grafana Labs.
```

Reply:
it's not really Grafana plugins, atm is only core plugins, and is only enabled for a hardcoded list. I'll rephrase this.